### PR TITLE
Ber errors

### LINKED ISSUE #77: start using error returning BER API calls
DecodePacketErr returns error information, we should update to use this and propagate errors

also, we should add bounds checking helpers that return a child packet or an error

would do this on a major release boundary since it would probably mean tweaking function return values

### PR DESCRIPTION
**Pull request by vetinari**
functions which were using ber.DecodePacket() now return an error also.

fixes #77

### CODE REVIEW

File: ldap.go [hunk 1]
```diff
@@ -116,37 +117,37 @@ func addLDAPDescriptions(packet *ber.Packet) (err error) {
...
+		err = addRequestDescriptions(packet)
 	case ApplicationExtendedResponse:
 	}
 
-	return nil
+	return err
 }
 
-func addControlDescriptions(packet *ber.Packet) {
+func addControlDescriptions(packet *ber.Packet) error {
```
Comment (liggitt) [inline ldap.go]: aren't descriptions advisory/optional? I'm trying to decide is failing to add them should be a fatal error

Comment (vetinari) [reply ldap.go]: Failing means the packet is not in a valid format. This is just called (directly or indirectly) from `addLDAPDescriptions` which returns an error on broken packets (e.g. not enough children in a sequence).

File: dn.go [hunk 2]
```diff
@@ -1,3 +1,7 @@
+// Copyright 2015 The Go Authors. All rights reserved.
```
Comment (liggitt) [inline dn.go]: Not sure where this copyright header came from

File: ldap.go (hunk 1 above)
Comment (liggitt) [reply ldap.go]: ok

**Comment by liggitt** [comment]
remove copyright header and squash, then lgtm

**Pull request merged**